### PR TITLE
Mk hackathon repeater edit

### PR DESCRIPTION
**Pull request by mkangia**
Started this earlier [here](https://github.com/dimagi/commcare-hq/pull/15430).
A lot has changed since then so doing a fresh start here.
[also listed under ideas](https://docs.google.com/spreadsheets/d/1OqNRh7DBxbvObML4Vk68T_Rye43zXCHG_Vv0GUnldQE/edit#gid=0&range=33:33)

Best review 🐟 / 🐟 

This PR
1. Adds the edit view for repeaters.
2. Ability to pause and then resume a repeater: 
This means that the repeat records would keep getting added for a repeater but would trigger only when resumed. The next check for such records is delayed by one hour at each check, to avoid clogging queue at the time of processing records with repeat records that are not be triggered.

UI changes:
1. Added edit view
2. Added two buttons in the repeaters list:  Pause, Resume and renamed "Stop Forwarding" to "Stop".
3. Added modal confirmation pop-ups for pause and resume and also added more text to "Stop" to share the records are not deleted but are just hidden from the list.

Since this could not be tested on staging put this behind a feature flag for now and then release it generally once tested on some live environment

### CODE REVIEW

**Comment by mkangia** [comment]
@sheelio please help confirm the correct label for this.

This adds edit for repeaters. These views are accessible to only domain admins. Also not sure how many domains use this but label product/all-users looked like this best possible label for this. Let me know if this should be changed.

**Comment by sheelio** [comment]
@mkangia 
I think product/all-users is the right label here.

**Review by dannyroberts — APPROVED** [review]
Awesome! I've certainly wanted this before

**Comment by mkangia** [comment]
@dannyroberts made some changes post your last review.

File: corehq/motech/repeaters/models.py [hunk 1]
```diff
@@ -284,6 +290,16 @@ def handle_response(self, result, repeat_record):
...
 
+    @property
+    def is_form_repeater(self):
+        # check if any of parent class is FormRepeater
+        return bool(filter(lambda parent_class: parent_class.__name__ == "FormRepeater", self.__class__.__mro__))
+
+    @property
+    def is_case_repeater(self):
+        # check if any of parent class is CaseRepeater
+        return bool(filter(lambda parent_class: parent_class.__name__ == "CaseRepeater", self.__class__.__mro__))
```
[the remark this answers is not in the document]
Comment (dannyroberts) [reply corehq/motech/repeaters/models.py]: does this line mean the same thing?

```
any(parent_class.__name__ == "CaseRepeater" for parent_class in self.__class__.__mro__)
```

If so, I think the `any` version is a lot more idomatic (and also aborts as soon as it finds a match, though any performance in this case would probably be imperceptible).

File: corehq/motech/repeaters/models.py [hunk 2]
```diff
@@ -284,6 +290,16 @@ def handle_response(self, result, repeat_record):
             self.generator.handle_failure(result, self.payload_doc(repeat_record), repeat_record)
         return attempt
 
+    @property
+    def is_form_repeater(self):
+        # check if any of parent class is FormRepeater
+        return bool(filter(lambda parent_class: parent_class.__name__ == "FormRepeater", self.__class__.__mro__))
```
[the remark this answers is not in the document]
Comment (dannyroberts) [reply corehq/motech/repeaters/models.py]: (ditto)

**Review by dannyroberts — APPROVED** [review]

File: corehq/motech/repeaters/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (esoergel) [reply corehq/motech/repeaters/models.py]: wouldn't `isinstance(self, CaseRepeater)` also do the trick?

[the remark this answers is not in the document]
Comment (dannyroberts) [reply corehq/motech/repeaters/models.py]: Oh, hah! Yeah what @esoergel said

[the remark this answers is not in the document]
Comment (mkangia) [reply corehq/motech/repeaters/models.py]: oh! I don't remember now why i did not use `isinstance(self, CaseRepeater)` and did that iteration. Will change to `isinstance(self, CaseRepeater)` for now.

**Comment by mkangia** [comment]
we could not QA this on staging since we don't have repeaters running on it since its a test environment. So i put this behind a FF and tested on local machine. Once QAed on a live environment can roll this out to all.
fyi @sheelio @dannyroberts

**Pull request merged**